addons/stock/models/stock_rule.py

```diff
@@ -35,36 +35,143 @@ class ProcurementRule(models.Model):
-    location_id = fields.Many2one('stock.location', 'Procurement Location')
-    location_src_id = fields.Many2one('stock.location', 'Source Location', help="Source location is action=move")
+    company_id = fields.Many2one('res.company', 'Company',
+        default=lambda self: self.env.user.company_id)
+    location_id = fields.Many2one('stock.location', 'Destination Location', required=True)
+    location_src_id = fields.Many2one('stock.location', 'Source Location')
     route_id = fields.Many2one('stock.location.route', 'Route', required=True, ondelete='cascade')
     procure_method = fields.Selection([
         ('make_to_stock', 'Take From Stock'),
-        ('make_to_order', 'Create Procurement')], string='Move Supply Method',
+        ('make_to_order', 'Trigger Another Rule')], string='Move Supply Method',
         default='make_to_stock', required=True,
-        help="""Determines the procurement method of the stock move that will be generated: whether it will need to 'take from the available stock' in its source location or needs to ignore its stock and create a procurement over there.""")
+        help="""Create Procurement: A procurement will be created in the source location and the system will try to find a rule to resolve it. The available stock will be ignored.
+             Take from Stock: The products will be taken from the available stock.""")
     route_sequence = fields.Integer('Route Sequence', related='route_id.sequence', store=True)
     picking_type_id = fields.Many2one(
         'stock.picking.type', 'Operation Type',
-        required=True,
-        help="Operation Type determines the way the picking should be shown in the view, reports, ...")
-    delay = fields.Integer('Number of Days', default=0)
-    partner_address_id = fields.Many2one('res.partner', 'Partner Address')
+        required=True)
+    delay = fields.Integer('Delay', default=0, help="The expected date of the created transfer will be computed based on this delay.")
+    partner_address_id = fields.Many2one('res.partner', 'Partner Address', help="Address where goods should be delivered. Optional.")
     propagate = fields.Boolean(
         'Propagate cancel and split', default=True,
-        help='If checked, when the previous move of the move (which was generated by a next procurement) is cancelled or split, the move generated by this move will too')
-    warehouse_id = fields.Many2one('stock.warehouse', 'Served Warehouse', help='The warehouse this rule is for')
+        help="When ticked, if the move is splitted or cancelled, the next move will be too.")
+    warehouse_id = fields.Many2one('stock.warehouse', 'Warehouse')
     propagate_warehouse_id = fields.Many2one(
         'stock.warehouse', 'Warehouse to Propagate',
         help="The warehouse to propagate on the created move/procurement, which can be different of the warehouse this rule is for (e.g for resupplying rules from another warehouse)")
+    auto = fields.Selection([
```

Comment:
is it possible to set auto on a pull rule? shouldn't we block this?

Reply:
Also hide in view